### ISSUE TITLE
bug(transforms): MixtureGaussianCDF.inverse fails for tail samples (too-narrow bisection bracket)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#108]
## Summary

`MixtureGaussianCDF.inverse` (and `inverse_and_log_det`) is **exact in the bulk but fails for tail samples** — the round-trip error `|x - inverse(transform(x))|` blows up to ~1.8 for points a few standard deviations out, even when the bijector is fit to the data with `from_data`. The sibling `MixtureLogisticCDF` and `RQSplineMarginal` invert the *same* data to ~1e-14, so this looks specific to the Gaussian-mixture inverse (most likely a too-narrow / non-adaptive bisection bracket that returns the bracket edge for tail targets).

This propagates to `gaussianization_flow`: its bijection round-trip (`z -> transform -> inverse`) errors at ~2.0, so `flow.sample` produces inaccurate tail samples (it runs without error, masking the problem).

## Repro

```python
import jax, jax.numpy as jnp, numpy as np
import gauss_flows as gf
jax.config.update("jax_enable_x64", True)

x = jnp.asarray(np.random.default_rng(0).standard_normal((2000, 2)))  # max|x| ~ 3.5

b = gf.MixtureGaussianCDF.from_data(x, 8)
z  = jax.vmap(b.transform)(x)
xr = jax.vmap(b.inverse)(z)
err = jnp.abs(x - xr).max(axis=1)
print("MixtureGaussianCDF : max", float(err.max()),
      "median", float(jnp.median(err)),
      "frac>1e-5", float((err > 1e-5).mean()))
# -> max 1.82, median 6.5e-14, frac>1e-5 0.071  (all failures at |x| > ~2.5)

bl = gf.MixtureLogisticCDF(n_components=8, shape=(2,))
zl = jax.vmap(bl.transform)(x)
print("MixtureLogisticCDF : max",
      float(jnp.abs(x - jax.vmap(bl.inverse)(zl)).max()))   # -> ~9e-14
```

Observed (jax 0.10.1, gauss_flows @ current main):

| bijector | max round-trip err | median | frac > 1e-5 |
|---|---|---|---|
| `MixtureGaussianCDF` (fitted) | **1.82** | 6.5e-14 | 7.1% (all tails) |
| `MixtureLogisticCDF` | 8.9e-14 | — | 0% |
| `RQSplineMarginal` | 3.3e-16 | — | 0% |

## Expected

`inverse(transform(x)) == x` to ~1e-10 for all in-distribution samples, matching the logistic/spline marginals.

## Likely fix

Widen / make adaptive the bisection bracket in the `MixtureGaussianCDF` inverse (bracket from the component means ± k·scales, or expand until the target is bracketed), and/or clamp the probit input consistently with the forward pass.

## Context

Found while writing a "numerical mechanics of bijectors" foundations tutorial (round-trip invertibility test) for the Gaussianization curriculum.
